Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -341,6 +342,43 @@ public IUploadProgress GetProgress()
 
         #endregion
 
+        #region UploadSessionData
+        /// <summary>
+        /// Event called when an UploadUri is created. 
+        /// Not needed if the application program will not support resuming after a program restart.
+        /// </summary>
+        /// <remarks>
+        /// Within the event, persist the UploadUri to storage.
+        /// It is strongly recommended that the full path filename (or other media identifier) is also stored so that it can be compared to the current open filename (media) upon restart.
+        /// </remarks>
+        public event Action<IUploadSessionData> UploadSessionData;
+        /// <summary>
+        /// Data to be passed to the application program to allow resuming an upload after a program restart.
+        /// </summary>
+        private class ResumeableUploadSessionData : IUploadSessionData
+        {
+            /// <summary>
+            /// Create a ResumeableUploadSessionData instance to pass the UploadUri to the client.
+            /// </summary>
+            /// <param name="uploadUri">The resumable session URI.</param>
+            public ResumeableUploadSessionData(Uri uploadUri)
+            {
+                UploadUri = uploadUri;
+            }
+            public Uri UploadUri { get; private set; }
+        }
+        /// <summary>
+        /// Send data (UploadUri) to application so it can store it to persistent storage.
+        /// </summary>
+        private void SendUploadSessionData(ResumeableUploadSessionData sessionData)
+        {
+            if (UploadSessionData != null)

```

Comment:
As for using Action - I think it was a mistake in the original, personally. But I guess it makes sense to be consistent now.